zerver/lib/home.py

```diff
@@ -35,6 +35,23 @@ class UserPermissionInfo:
     show_webathena: bool
 
 
+def is_outdated_server(
+    server_upgrade_nag_deadline: int, last_server_upgrade_time: int
+) -> Tuple[bool, bool]:
+    # Should be invoked with the SERVER_UPGRADE_NAG_DEADLINE
+    # And LAST_SERVER_UPGRADE_TIME settings as parameters
+    # Returns (needs_upgrade, insecure)
+
+    # Users are allowed to override the default SERVER_UPGRADE_NAG_DEADLINE setting
+    # So if SERVER_UPGRADE_NAG_DEADLINE is overriden to a value < 365 reset to 365
+    if server_upgrade_nag_deadline < 365:
+        server_upgrade_nag_deadline = 365
```

Comment:
I don't understand why we want this block here?  

Also, you don't need the `server_upgrade_nag_deadline` parameter; you can use `with self.settings` to change the value of a setting for tests.  And I think it'd be simpler to have this function only return one boolean, and accept the UserProfile as a parameter (to check if they're an admin); we don't want to copy the exact interface for the desktop code.

(`git grep 'with self.settings'` will show how you to do that).

Reply:
Why do we need that block of code?
I was thinking since server admins could override the default server nag deadline(365 days), Even though very unlikely, I imagine that they might override it to a value less than the default nag deadline. In situations like that, that block of code is meant to reset the value to the initial nag deadline.

I understand the changes you suggested about removing the server_upgrade_nag_dealine parameter and returning just one boolean. I'd integrate the changes.

One thing, I don' understand the part about checking if a user is an admin. Why do we need to do that?

Reply:
I don't think we want that -- in particular, for tests, we may well want to set the deadline to 0.

You can ignore the admin thing for now -- it was an extension idea in the original issue.